### ISSUE TITLE
[rush][bug] rushx command exits with "An unknown error occurred" message with rush version 5.19.1

### ISSUE THREAD

**Issue by KarishmaGhiya** [microsoft/rushstack#1723]
**Is this a feature or a bug?**

- [ ] Feature
- [x ] Bug

**Please describe the actual behavior.**

The rushx command exits with a nonzero code and with the message - "An unknown error occurred"
The latest rush version 5.19.1 has this bug. The earlier version 5.17.2 doesn't  have this bug.

**If the issue is a bug, how can we reproduce it? Please provide detailed steps and include a GitHub branch if applicable. Your issue will get resolved faster if you can make it easy to investigate.**

You can reproduce it by running the rushx command inside any package. Eg - you can use our mono-repo

```
git clone https://github.com/Azure/azure-sdk-for-js.git
git checkout dcd57274dbe9acd7deb3d34173a9582017f71203
cd azure-sdk-for-js
rush install
rush build
cd sdk/servicebus/service-bus 
[OR ANY PACKAGE]
rushx build
rushx unit-test
rushx lint
```
**What is the expected behavior?**

You will notice that all the three commands run - build, lint and unit-test executed with rushx do produce results as expected. but they exit with a non-zero exit code with the error message - "An unknown error occurred" without any further information about the error even when we run it with `--verbose`

In this code for rush - we are probably missing out a case where there is successful execution yet it is returning an error - https://github.com/microsoft/rushstack/blob/master/apps/rush-lib/src/scripts/install-run.ts#L436
May be try != null instead of !== to also cover `undefined ` scenario
```
if (result.status != null) {
    return result.status;
  }
  else {
    throw result.error || new Error('An unknown error occurred.');
  }
```
instead of 
```
if (result.status !== null) {
    return result.status;
  }
  else {
    throw result.error || new Error('An unknown error occurred.');
  }
```

**Comment by mikeharder** [comment]
This also repros in https://github.com/microsoft/rush-example when upgraded to `5.19.1`:

```
root@eca61a3819c1:/rush-example/apps/my-app# rushx build
Found configuration in /rush-example/rush.json


Rush Multi-Project Build Tool 5.19.1 - https://rushjs.io
Node.js version is 10.18.1 (LTS)

Found configuration in /rush-example/rush.json

Executing: "node_modules/.bin/my-build"

Invoking my-toolchain...
Success!
Error: An unknown error occurred.
```

**Comment by mikeharder** [comment]
@iclanton: This might already be fixed by #1722.  If so, can you please release `5.19.2` with this PR?

**Comment by iclanton** [comment]
@mikeharder - sure thing. This has been released in `5.19.2`

**Comment by mikeharder** [comment]
@iclanton: Thanks for the quick turnaround, but that version appears to be broken:

https://github.com/microsoft/rushstack/issues/1725
